### DISCUSSION TITLE
swift-dependencies doesn't compile with Xcode 16.0 beta 6 (16A5230g)

### DISCUSSION THREAD

**Discussion by rurza** [pointfreeco/swift-dependencies#260] [Q&A] [answered]
Today, I updated Xcode to the latest beta version, and now I can’t compile my project. The compilation fails in the WithDependencies file on line 71.

```
Returning a task-isolated '@noescape @async @callee_guaranteed @substituted <τ_0_0> () -> (@out τ_0_0, @error any Error) 
for <R>' value as a 'sending' result risks causing data races
```

Can someone confirm the issue?

**Comment by mbrandonw** [comment]
Hi @rurza, I haven't downloaded beta 6 yet, but can you please confirm which version of Dependencies you are using? There have been multiple fixes to this problem throughout the beta period.

**Reply by rurza** [reply]
1.3.7

**Reply by mbrandonw** [reply]
Ok thanks, once I can download beta 6 I will confirm.

**Reply by rurza** [reply]
In the meantime, I've tested it on Xcode **16.1** beta (16B5001e), and it does compile

**Reply by rayfix** [reply]
FWIW, same error with swift-dependencies 1.3.7 and the Sequoia Xcode release. Xcode 16 Version 16.0 (16A242d) Happens with Swift 5 and Swift 6 mode, with different levels of concurrency checking.

**Reply by rayfix** [reply]
Problem does not occur in 1.4.0

**Reply by mbrandonw** [reply, marked as answer]
Hi @rayfix, if you check out the [release notes](https://github.com/pointfreeco/swift-dependencies/releases) you will see that 1.3.8, 1.3.9 and 1.4.0 all fixed various problems during the beta period. Now that the final version of Xcode 16 is out we believe that the newest release should have no build errors.

**Reply by rayfix** [reply]
Thank you!
